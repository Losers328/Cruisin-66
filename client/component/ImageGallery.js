import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableWithoutFeedback,
  Dimensions,
  Modal,
  Image
} from 'react-native';

// import ImageElement from './ImageElement.js';

export default class ImageGallery extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modalVisible: false,
      modalImage: require('./images/amy.jpg'),
      images: [
        require('./images/amy.jpg'),
        require('./images/couple1.jpg'),
        require('./images/couple2.jpg'),
        require('./images/miller1.jpg'),
        require('./images/miller2.jpg'),
        require('./images/miller3.jpg'),
        require('./images/pusheen.jpg'),
        require('./images/stormy1.jpg'),
        require('./images/stormy2.jpg'),
      ]
    }
  }

  setModalVisible(visible, imageKey) {
    this.setState({
      modalImage: this.state.images[imageKey],
      modalVisible: visible
    });
  }

  getImage() {
    return this.state.modalImage;
  }

  render() {
    let images = this.state.images.map((image, key) => {
      return <TouchableWithoutFeedback key={key}
                onPress={() => { this.setModalVisible(true, key)}}>
                <View style={styles.imagewrap}>
                  <Image source={image} style={styles.image}></Image>
                </View>
              </TouchableWithoutFeedback>
    })
    return (
      <View style={styles.container}>
        <Modal style={styles.modal} 
          animationType={'fade'}
          transparent={true}
          visible={this.state.modalVisible}
          onRequestClose={() => {}}>
          <View style={styles.modal}>
            <Text style={styles.text}
              onPress={() => {this.setModalVisible(false)}}>
              Close
            </Text>
            <Image source={this.state.modalImage}></Image>
          </View>
        </Modal>
        {images}
      </View>
    );
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    backgroundColor: '#eee'
  },

  imagewrap: {
    margin: 2,
    padding: 2,
    height: (Dimensions.get('window').height / 3) - 12,
    width: (Dimensions.get('window').width / 2) - 4,
    backgroundColor: '#fff',
  },

  image: {
    flex: 1,
    width: null,
    alignSelf: 'stretch',
  },

  modal: {
    flex: 1,
    padding: 40,
    backgroundColor: 'rgba(0, 0, 0, 0.9)'
  },

  text: {
    color: '#fff',
  }
});