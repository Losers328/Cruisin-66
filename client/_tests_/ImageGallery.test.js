import React from 'react';
import ImageGallery from '../component/ImageGallery.js';
import renderer from 'react-test-renderer';


describe('Image Gallery', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ImageGallery/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});