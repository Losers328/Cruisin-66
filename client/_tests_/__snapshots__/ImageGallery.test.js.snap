// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Gallery renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.9)",
        "flex": 1,
        "padding": 40,
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.9)",
          "flex": 1,
          "padding": 40,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Close
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../client/component/images/amy.jpg",
          }
        }
      />
    </View>
  </Modal>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/amy.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/couple1.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/couple2.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/miller1.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/miller2.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/miller3.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/pusheen.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/stormy1.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 432.6666666666667,
        "margin": 2,
        "padding": 2,
        "width": 371,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../client/component/images/stormy2.jpg",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "width": null,
        }
      }
    />
  </View>
</View>
`;
